### PR TITLE
Improve C++ stub publishing

### PR DESCRIPTION
**Pull request by gavinreaney**
Note: cmake is required on the host machine to do package creation and publishing.

1) Bintray user and API key are read from gradle properties
2) CMakeLists use glob to find source files.
3) Only publish non-snapshot versions
4) Version number is automatically populated into conanfile.py via templating

### CODE REVIEW

File: stubs/cpp/build.gradle [hunk 1]
```diff
@@ -52,26 +52,49 @@ protobuf {
 
 }
 
+task prepareConanfilePy {
+    inputs.property("version", version);
```
Comment (anuraaga) [inline stubs/cpp/build.gradle]: Let's use single quotes for non-template strings.

File: stubs/cpp/build.gradle [hunk 2]
```diff
@@ -52,26 +52,49 @@ protobuf {
...
+    outputs.file("$buildDir/generated/scripts/conanfile.py")
+
+    doFirst {
+        def template = file('package/conanfile.py.tmpl').text
+        def filled = template.replaceFirst('\\|API_VERSION\\|', version.toString())
+        def outDir = "$buildDir/generated/scripts"
+        mkdir outDir
+        def out = file("${outDir}/conanfile.py")
+        out.text = filled
+        out.executable = true
```
Comment (anuraaga) [inline stubs/cpp/build.gradle]: Don't think conanfiles need to be executable

File: stubs/cpp/build.gradle [hunk 3]
```diff
@@ -84,4 +107,33 @@ afterEvaluate {
     tasks.generateProto.doLast {
         delete 'build/conan/src/google'
     }
+
+    def conanDir = 'build/conan'
+    def reference = 'stellarstation/testing'
```
Comment (anuraaga) [inline stubs/cpp/build.gradle]: Let's go ahead and use stable

File: stubs/cpp/build.gradle [hunk 4]
```diff
@@ -84,4 +106,33 @@ afterEvaluate {
...
+        dependsOn createPackage
+
+        doLast {
+            exec {
+                def user = rootProject.property('bintray.user')
+                def key = rootProject.property('bintray.key')
+
+                executable 'bash'
+                args '-c', condaCommand(
+                        "conan user -p $key -r stellarstation $user && " +
```
Comment (anuraaga) [inline stubs/cpp/build.gradle]: I can't find this in my bash history or even setting environment variables, but I think we can avoid two steps by using variables

https://docs.conan.io/en/latest/reference/env_vars.html

Comment (gavinreaney) [reply stubs/cpp/build.gradle]: I switched to using env vars, but it seems to end up running 'conan user' anyway. (Conan detects the failure to authenticate, then gets the details from the env and runs conan user).

**Pull request merged**